bundles/org.openhab.binding.speedtest/src/main/java/org/openhab/binding/speedtest/internal/SpeedtestHandler.java

```diff
@@ -331,8 +349,19 @@ private void getSpeed() {
      */
     private void updateChannels() {
         logger.debug("Updating channels");
+        State newState;
 
-        State newState = new QuantityType<>(Double.parseDouble(pingJitter) / 1000.0, Units.SECOND);
+        // timestamp format: "2023-07-20T19:34:54Z"
+        try {
+            ZonedDateTime zonedDateTime = ZonedDateTime.parse(timestamp).withZoneSameInstant(ZoneId.systemDefault());
+            newState = new DateTimeType(zonedDateTime);
+            logger.debug("timestamp: {}", newState);
+            updateState(new ChannelUID(getThing().getUID(), SpeedtestBindingConstants.TIMESTAMP), newState);
+        } catch (DateTimeParseException e) {
+            logger.debug("Exception: {}", e.getMessage());
+        }
+
+        newState = new QuantityType<>(Double.parseDouble(pingJitter) / 1000.0, Units.SECOND);
```

Comment:
Consider using different variable names for each state. That makes further PRs more understandable.

Reply:
reworked together with the above change